### DISCUSSION TITLE
package.json problem sowie no debugger available

### DISCUSSION THREAD

**Discussion by Vince2312** [VDI-CodING/codeING-main#39] [Q&A]
Wenn ich versuche Programmcode zu kompilieren, wirft die Konsole folgende Probleme. Habe nochmal das Video nachgeschaut, konnte das Problem aber bisher nicht lösen.

<img width="315" alt="Compiler Problem" src="https://user-images.githubusercontent.com/80846982/111771191-ac834000-88ab-11eb-8228-4ccfed1d48b3.PNG">

Wo müsste ich "type": "module" denn einfügen in der package.json Datei?

**Comment by KlausLex** [comment]
`S1D0 - Einrichtung` hat mit einfügen der Skripte in `package.json` geendet, da war ja noch gar nichts zu kompilieren?
Einfach speichern und via Terminal `npm run start-dev` ausführen.

Dann kommt die Ausgabe `Server lauscht auf 5000` und im Browser unter `localhost:5000` siehst du `Hallo Welt`

**Reply by Vince2312** [reply]
Das hat auch geklappt. Ich meine S1D1 Aufgabe 1

**Reply by Zangarus** [reply]
Du musst die Skripte nutzen, um das Programm zu kompilieren. Ein einfacher Aufruf der Datei `.\src\app.ts` über die Kommandozeile führt leider zu nichts, da unser Computer nicht weiß, wie man diese kompilieren kann.

`npm run start-watch` oder `npm run start-dev`sollte zum Ausführen des Programms genutzt werden. Die Ergbenisse siehst du dann wahrscheinlich in der Kommandozeile, wenn du dir bestimmte Sachen im Code mit `console.log()`ausgeben lässt.

**Reply by w-tusche** [reply]
Hier ist noch etwas was du probieren kannst falls du eine neue Datei S1D1.ts erstellt hast/erstellen möchtest:

Auf die Datei im VS-Explorer einen Rechtsklick machen > Open in Integrated Terminal und da gibst du dann `tsc S1D1.ts` oder mit `npx tsc S1D1.ts` ein, dann wird die TypeScript Datei zu einer JavaScript Datei compiliert. Diese kannst du dann mit `node S1D1.js` ausführen.
